### PR TITLE
Avoid vulnerable versions of `ws` in dependency tree

### PR DESCRIPTION
**Pull request by mtlewis**
## Hey, I just made a Pull Request!

This PR aims to mitigate [SNYK-JS-WS-1296835](https://snyk.io/vuln/SNYK-JS-WS-1296835) by avoiding vulnerable versions of ws in our yarn.lock file. I took two different actions here, both of which have the potential to be controversial, so keen to get feedback from maintainers:

1. Removed the `ws` entries from yarn.lock and re-ran `yarn install`, committed as 1401985. This has the effect of updating the version of `ws` used to a non-vulnerable one everywhere in the tree except for one place (see 2.)
2. Used a resolution to force the version of `ws` used inside `@graphql-tools/url-loader` to 7.4.6. The version of `ws` is currently pinned in this package, which means there's no way to avoid the vulnerable version with yarn.lock changes alone.

I don't personally know whether fixes in yarn.lock alone will be enough to make Snyk happy. Since projects that depend on Backstage packages will have their own entirely separate yarn.lock file, it's still entirely possible for integrators to be running versions of `ws` that have been marked by Snyk regardless of the contents of our lockfile.

Once we have Snyk monitoring this repository directly, it will be much easier to verify fixes. Note that integrating directly with Snyk is currently blocked - I just opened https://github.com/snyk/snyk/issues/2126 which describes the blocking issue.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     ]
   },
   "resolutions": {
+    "**/@graphql-tools/url-loader/**/ws": "7.4.6",
```
Comment (OrkoHunter) [inline package.json:44]: Maybe this, to continue receiving patch updates? :)

```suggestion
    "**/@graphql-tools/url-loader/**/ws": "^7.4.6",
```

**Review by OrkoHunter — COMMENTED** [review]
The resolution looks fine to me, bummer that we can't add a comment in JSON for why that resolution is needed.

Although, looking at the package's dependency, they seem to be using version ^8.0.0 of ws 🤔  https://github.com/ardatan/graphql-tools/blob/7ad668781cd7b61300b448e9061db4b96bc8266a/packages/loaders/url/package.json#L65

**Comment by benjdlambert** [comment]
Can we just bump the `graphql-tools` dependency entirely?

File: package.json (hunk 1 above)
Comment (mtlewis) [reply package.json:44]: Makes sense!

**Comment by mtlewis** [comment]
> Although, looking at the package's dependency, they seem to be using version ^8.0.0 of ws 🤔 https://github.com/ardatan/graphql-tools/blob/7ad668781cd7b61300b448e9061db4b96bc8266a/packages/loaders/url/package.json#L65

> Can we just bump the graphql-tools dependency entirely?

Unfortunately not - the dependency chain is `@backstage/plugin-catalog-graphql` > `@graphql-codegen/cli` -> `@graphql-tools/url-loader` -> `ws`. Until `@graphql-codegen/cli` is upgraded to use the latest `@graphql-tools` packages, I think we're stuck with resolutions. I did take a quick look at the `@graphql-codegen/cli` repo - it looks like a non-trivial upgrade, especially since it probably makes sense to upgrade all the `@graphql-tools` packages at the same time. I [posted a comment](https://github.com/dotansimha/graphql-code-generator/issues/6274) to try to get this upgrade back on their radar though - looks like last time it was discussed, `@graphql-tools` hadn't yet been updated.

File: package.json (hunk 1 above)
Comment (mtlewis) [reply package.json]: Done :)

**Comment by mtlewis** [comment]
@OrkoHunter 

> Although, looking at the package's dependency, they seem to be using version ^8.0.0 of ws 🤔 https://github.com/ardatan/graphql-tools/blob/7ad668781cd7b61300b448e9061db4b96bc8266a/packages/loaders/url/package.json#L65

Yep, so url-loader has already been upgraded to avoid the problematic dependency. The root of the problem is the fact that `@graphql-codegen/cli` needs to be upgraded to use the new `graphql-tools` packages. Perhaps switching to including that package in the resolution might make that fact clearer?

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**